flexmeasures/utils/plugin_utils.py

```diff
@@ -108,24 +100,75 @@ def register_plugins(app: Flask):
-        if missing_required_config_settings:
-            raise ConfigurationError(
-                f"Missing required config setting(s): {missing_required_config_settings}"
-            )
+def check_config_settings(app, settings: Dict[str, dict]):
+    """Make sure expected config settings exist.
+
+    For example:
+
+        settings = {
+            "MY_PLUGIN_URL": {
+                "description": "URL used by my plugin for x.",
+                "level": "error",
+            },
+            "MY_PLUGIN_TOKEN": {
+                "description": "Token used by my plugin for y.",
+                "level": "warning",
+                "message": "Without this token, my plugin will not do y.",
+                "parse_as": str,
+            },
+            "MY_PLUGIN_COLOR": {
+                "description": "Color used to override the default plugin color.",
+                "level": "info",
+            },
+        }
+
+    """
+    assert isinstance(settings, dict), f"{settings} should be a dict"
+    for setting_name, setting_fields in settings.items():
+        assert isinstance(setting_fields, dict), f"{setting_name} should be a dict"
+
+    missing_config_settings = []
+    config_settings_with_wrong_type = []
+    for setting_name, setting_fields in settings.items():
+        setting = app.config.get(setting_name)
+        if setting is None:
+            missing_config_settings.append(setting_name)
+        elif "parse_as" in setting_fields and not isinstance(
+            setting, setting_fields["parse_as"]
+        ):
+            config_settings_with_wrong_type.append((setting_name, setting))
+    for setting_name, setting in config_settings_with_wrong_type:
+        log_wrong_type_for_config_setting(
+            app, setting_name, settings[setting_name], setting
+        )
+    for setting_name in missing_config_settings:
+        log_missing_config_setting(app, setting_name, settings[setting_name])
+
+
+def log_wrong_type_for_config_setting(
+    app, setting_name: str, setting_fields: dict, setting: Any
+):
+    """Log a message for this config setting that has the wrong type."""
+    app.logger.warning(
+        f"Config setting '{setting_name}' is a {type(setting)} whereas a {setting_fields['parse_as']} was expected."
+    )
+
+
+def log_missing_config_setting(app, setting_name: str, setting_fields: dict):
+    """Log a message for this missing config setting.
+
+    The logging level is taken from the 'level' key. If missing, we default to error.
+    If present, we also log the 'description' and the 'message_if_missing' keys.
+    """
+    message_if_missing = (
+        f" {setting_fields['message_if_missing']}"
+        if "message_if_missing" in setting_fields
+        else ""
+    )
+    description = (
+        f" ({setting_fields['description']})" if "description" in setting_fields else ""
+    )
+    level = setting_fields["level"] if "level" in setting_fields else "error"
+    getattr(app.logger, level)(
```

Comment:
Is `level`  case-sensitive? We can also check if a wrong level was passed (`hasattr(app.logger, level)`). If so, warn about that and fall back to 'warning', I guess.

Reply:
Good suggestion. I added the warning for the wrong level and chose to fall back to 'error' for the missing config.